### DISCUSSION TITLE
Ordering 3d prints online: ~1mm wall thickness, making version with brass inserts for resin printing, etc.

### DISCUSSION THREAD

**Discussion by evantravers** [juhakaup/keyboards#23] [General]
All my local 3d printing friends have their printers down for the count. I've been looking at ordering the Cygnus from pcbway or similar, but they have a policy where they won't print less than 1mm with FDM printing or something.

I have considered getting them to resin print the case, but then I'd need to articulate where the brass screw inserts go in the model for them to handle.

I guess I'm wondering:

1. Has anyone found a company that'll print the cygnus case online?
2. I wonder if it's worth making a parted version with the brass inserts in the model for resin printing using an online service.

**Comment by evantravers** [comment]
I'm still working on this… it seems that I have two choices:

1. Ask pretty please and accept the fact that the parts are too thin.
2. Learn some CAD skills and add the brass inserts as parts so that I could resin print it.

Currently pursuing 1., will report back.

**Comment by juhakaup** [comment]
I don't really understand the brass inserts part. Can't you just add them yourself when you get the parts?
What I understand is that you could just glue the inserts in place, as you can't melt the resin.

**Comment by evantravers** [comment]
It's likely that _I_ don't understand @juhakaup.

I'm planning on getting them to FDM printing and adding them myself as you demo'd in your lovely video.

In talking back and forth with their support, you can resin or nylon SLS the model, and they have a place in their order form for specifying inserts:
![image](https://github.com/juhakaup/keyboards/assets/62209/8ff6eba9-8763-41eb-9a8c-c3b0d970e379)

I believe they can add them to nylon SLS and Resin, so that's interesting.

I have in my head that a clear resin Cygnus would be pretty cool looking.

**Comment by juhakaup** [comment]
I believe you can skip that part if you are going to add the inserts yourself. 
But It's nice that they have an option to have the inserts added by them, if you don't have the tools to do it yourself.

If you're going to get a resin print, you may need to modify the inserts a bit before gluing them in place as they are a bit oversized to fit the holes.

**Comment by LarssonMartin1998** [comment]
I didn't even realize that the inserts might be an issue when printing it in resin!

I got the same response from jlc3dp when I tried to print it in Nylon, that they would have me accept a risk before they can print it. I wrote about it in this issue: #25 

@juhakaup Have you ever had any issues with those areas marked in that issue?

**Reply by evantravers** [reply]
I'm _terrible_ at printing, and I've printed 3 cases, and none of them have failed at the points that the online vendors have pointed out.

**Reply by LarssonMartin1998** [reply]
Thanks a lot, I also got a very valuable answer in my issue that I created if you want to read it, I'll go ahead and order it. Thanks for commenting on my issue orignally!

**Comment by evantravers** [comment]
My prints are on there way… will report back.

**Comment by LarssonMartin1998** [comment]
Just received my two prints, they for the most part have no issues. It’s an extremely minor issue on one of the four bottoms (I ordered 2 sets, one for home and one for work).

Those carved out sections that are thinner have potruded like 0,2mm, meaning the shape is clearly visiblr from underneath. If you put light preassure, each of the bottoms flex easily on those spots, but it really isnt an issue.

that said though, I’d recommend anyone with the skills or the owner to just add 1mm to the entire base, that would make it much more sturdy and at less risk to break when printing.

**Comment by juhakaup** [comment]
One thing I could do, would be to rise the relieved columns by a mm.
At least for me, this would be a simpler solution than adding thickness to the base.
Yes I could hack it in a few minutes, but I'm trying to keep the models somewhat maintainable.

The releaves allow me to keep the height of the case low, which I'm really not that keen to change. 
I haven't had any issues with my own prints.

I appreciate you for sharing your experience. I don't have any experience with these external manufacturers, and therefore can't really give any advice on this subject.

**Comment by evantravers** [comment]
I don't think that the base thickness needs to be changed… I think it's a
good tradeoff.

**Comment by LarssonMartin1998** [comment]
That sounds like a great solution, much better than mine!

**Comment by evantravers** [comment]
The JLCPCB prints have arrived… and are pretty good quality. Some jumps here and there.

![IMG_3907](https://github.com/juhakaup/keyboards/assets/62209/4da1ff10-3fc2-47bf-8fd5-c2d754279995)

I'd once again recommend finding a local printer because of the cost of shipping… but it's a viable option if someone can't find one near them.

**Reply by dudicoco** [reply]
Nice! Is that FDM?

**Reply by dudicoco** [reply]
@evantravers?

**Reply by evantravers** [reply]
Sorry @dudicoco. Got lost in my messages. Yes. It's FDM from JLCPCB. They are a bit messy. I've seen better quality from hobbyists, but they did warn me their resin would be better.

**Comment by LarssonMartin1998** [comment]
Hey, I’ve received mine as well from JLCPCB. And initially it seemed good. I paired the print with all of the aliexpress links for the non wireless parts.

This is my first time ever soldering and working with anything 3D printed, so this is entirely possible and probably likely to be skill issues, but I thought that I’d bring it up non the less.

It seems like the MCU doesn’t fit into the USB-C hole in the bottom case for the left half of the keyboard. The female USB-C on the MCU isn’t centered, so mirroring the half doesn’t seem like good enough for me. Additionally, the TRSS sockets don’t fit into the hole, the round part barely goes in, but there is absolutely no room for pushing the square part of the TRSS into the hole, which results in it not being enough space for the final part that holds the MCU and TRSS in place (the one that you screw into)

Lastly, this definitely comes down to skill issues, however, how would I go about soldering the keyboard matrix on the right half of the keyboard? The row and column are the same I take it. However, do I connect it to the MCU just like the other half? And do I use the same holes for the wires on the PCBs even though the location of the MCU is inversed but the PCBs arent?

Not really sure what to do about it now, but hopefully I figure something out, also got two prints and parts for two boards so I could bring one to work, and it would suck if I am unable to make it work, especially with like $250 in just shipping and import taxes.

EDIT: I wonder if the TRSS issue is simply because mine is printed in Nylon instead of the plastic shown in the build guide? It might be too stiff do press it in.

**Reply by juhakaup** [reply]
I’m sorry to hear about your troubles, but I’m sure we can make this work.

Let’s begin with the wiring of the MCUs. The wiring is basically the same for the both sides. So the TGB and the YHN -columns are connected to the pin 20 on each controller, RFV/UJM to the pin 19 and so on. The rows are the same for both sides.

The controllers usb port placement is likely my fault, I just linked the seller that I’ve ordered a couple times. Maybe the board they sell has changed, or I just haven’t noticed since I’ve been building mostly wireless boards lately. Getting these controllers from china is kind of a wild west, the dimensions and ports placements can vary. Maybe I should just recommend some known manufacturers, although they are quite a bit more expensive.
You can of course get another controller, but depending on how bad the offset is, another option is to modify the case a bit. It’s plastic, so you can use an x-acto-knife or small file to remove a bit of the plastic to make the controller fit.

The TRRS sockets fit quite tight, and probably vary a bit with different materials. Just make sure there isn’t anything left in the hole from manufacturing. Enlarging the hole is probably not that easy, but maybe you can shave off a bit from the socket by using some fine grit sanding paper.

This is certainly not the easiest thing to build, especially if you have no previous experience, but I’m sure you can make it work.

**Reply by LarssonMartin1998** [reply]
Thanks for the tips! I’m away on vacation for a few days now, but I will try this when I get back and see if I can make it work, I’ll update later and post some pics of the final builds later!

**Reply by LarssonMartin1998** [reply]
I finally got some time over to work on the keyboard. After a lot of effort, time, and mistakes I managed to complete the build.

I craved the usb hole slightly to make room for the female connector in the MCU, and it was some extra stuff left from manufacturing in the TRSS hole like you said, it fit perfectly after removing it. Thanks for the tips, really helped me out.

Here is a picture of the finalized build:
![image0](https://github.com/user-attachments/assets/882ecf45-dc6f-42ef-aab3-7cb25c841904)

**Reply by dudicoco** [reply]
Nice! Where did you get the wrist rests from?

**Reply by LarssonMartin1998** [reply]
Got it from [amazon](https://www.amazon.com/gp/aw/d/B08LLDKF4P?psc=1&ref=ppx_pop_mob_b_asin_title). Thought the quality would be bad. However, it’s pretty good actually, not worse than any other typical wooden rest I’ve used from custom keyboard retailers

**Comment by TrialOrError** [comment]
Guys I'm relatively new to github. Which STL files should i drop in a 3d print ordering website/3d printer SD card? a little lost as some of them seem to have things that need to be fixed as a description.

**Reply by maxyurk** [reply]
@TrialOrError it depends on which variant you want to print.
You can browse the STLs here: https://github.com/juhakaup/keyboards/tree/main/Cygnus%20v1.0
<img width="771" alt="Screenshot 2024-09-02 at 14 23 10" src="https://github.com/user-attachments/assets/2d9bd99c-f66f-44d0-ad31-977a6c18c123">

choose the variant - 3x5 or 3x6 or 4x6. 
For each hand you need:
one "top" (few variants to choose from), one "bottom" and one "controller holder" (few variants to choose from).
You need to mirror all for the other hand.

**Reply by TrialOrError** [reply]
Thanks a lot!

**Comment by TrialOrError** [comment]
I got the PCBs from JLCPCB. They warned me that the thin parts could've
snapped off but it didn't at all.
Very happy how the shipping got handled
